internal/schema/schema.go

```diff
@@ -74,20 +79,32 @@ func NewEphemeral(resolver Resolver) Manager {
 	return mgr
 }
 
-func (m *manager) LoadSchema(ctx context.Context, url string) (*jsonschema.Schema, error) {
-	entry, ok := m.cache.Get(url)
+func (m *manager) LoadSchema(ctx context.Context, schemaURL string) (*jsonschema.Schema, error) {
+	entry, ok := m.cache.Get(schemaURL)
 	if ok {
 		return entry.schema, entry.err
 	}
 
 	e := &cacheEntry{}
-	e.schema, e.err = m.loadSchemaFromStore(ctx, url)
+	var notFoundErr *notFoundErr
+	if e.schema, e.err = m.loadSchemaFromStore(ctx, schemaURL); e.err != nil && errors.As(e.err, &notFoundErr) {
+		parsedSchemaURL, err := url.Parse(schemaURL)
+		switch {
+		case err != nil:
+			e.err = fmt.Errorf("failed to parse URL: %w", err)
+		case notFoundErr.Scheme != URLScheme:
+			e.err = fmt.Errorf("schema file %q does not exist", schemaURL)
+		case strings.TrimPrefix(parsedSchemaURL.Path, "/") != strings.TrimPrefix(notFoundErr.Path, util.SchemasDirectory+string(os.PathSeparator)):
+			e.err = fmt.Errorf("schema file %q referenced by the schema does not exist in the store", notFoundErr.Path)
+		default:
+			e.err = fmt.Errorf("schema file %q does not exist in the store", notFoundErr.Path)
```

Comment:
If you change `notFoundErr` to return the full URL it was trying to load and the resolved system path (e.g. `_schemas/foo/bar.json`) you can simplify this as follows.

```suggestion
        if notFoundErr.url != schemaURL {
            e.err = fmt.Errorf("schema %s referenced by %s doesn't exist", notFoundErr.fullPath, schemaURL)
        } else {
            e.err = fmt.Errorf("schema %s doesn't exist", notFoundErr.fullPath)
        }	
```